runtime/str.go

```diff
@@ -753,6 +753,12 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 					return nil, raised
 				}
 				val = s.Value()
+			} else if matches[7] == "o" {
+				if o.isInstance(LongType) {
+					val = toLongUnsafe(o).Value().Text(8)
```

Comment:
So I guess your implementation is consistent with CPython and we should leave it as is. Will approve and merge.